### PR TITLE
#1008, #1670 & #1709 - Connection Status Indicators

### LINKED ISSUE #1008: Indicate inactive contacts & groups in fields in tiles
When there are people (and groups) in a field, they are all displayed the same, making it hard for the user to distinguish which contact is active, paused, closed etc.

One possible and simple way to help identify the differences would be to have the word and box around the name be a different colour.

### LINKED ISSUE #1670: Typeaheads hide or grey out closed/archived/inactive
Most Bob's are closed, but i don't know which ones.
Let's add that visibility to the typeahead results.

image

### LINKED ISSUE #1709: Status to Advance Search
Add record status to advance search to help filter results

### PR DESCRIPTION
**Pull request by kodinkat**
fixes #1008 
fixes #1670
fixes #1709 

Screenshot 2022-06-23 at 16 48 26

Screenshot 2022-06-23 at 16 49 59

Screenshot 2022-06-23 at 16 50 21

Screenshot 2022-06-28 at 12 02 16

### CODE REVIEW

File: dt-posts/posts.php [hunk 1]
```diff
@@ -2470,6 +2470,19 @@ public static function get_all_connected_fields_on_list( $field_settings, &$reco
             $connected_posts_mapped[$cp["ID"]] = $cp;
         }
 
+        /**
+         * Get meta values for connection post record
+         */
+        //phpcs:disable
+        //WordPress.WP.PreparedSQL.NotPrepare
+        $connected_post_meta = $wpdb->get_results( "
```
[the remark this answers is not in the document]
Comment (kodinkat) [reply dt-posts/posts.php:2478]: Used to capture connection's record metadata; namely, current status value -> this is then used in the frontend when listing (displaying relevant status info) previously saved connections.

[the remark this answers is not in the document]
Comment (kodinkat) [reply dt-posts/posts.php]: __UPDATE:__ Logic has now been removed.

**Comment by squigglybob** [comment]
Firstly, just to say, that I'm loving having the colours of the status's in so many places, it gives some really valuable information to the viewer without using up too much space, and makes it look prettier too :)

My first thought is that with the colours in all of these typeaheads and dropdowns and such, the contact list is looking a little drab. Knowing that this PR is adding status colours in different places, my first thought was to look at the contact list, but there was no colour 😞 .

Are there any thoughts to add status colours to the contacts list as well? I know we already have a column which shows by default for the status names for each record, but if it could also be coloured in some way that would also give a nice visual cue on the list. Just a thought.

**Comment by squigglybob** [comment]
Secondly, I have a question about the "load more" of the advanced search.  When clicking around to see what did what, I noticed a load more button on the record title section.

Clicking the button added the next set of contacts to the list but at the top...
image

where Alsha is the first contact that was on the list before I pressed the button, and the next set of contacts have been prepended to the list in reverse alphabetical order.

Would it make more sense for the load more to be at the bottom of the e.g. contacts record section?
And then when you click it, it will be adding the new records to the bottom of the list as well where you would be expecting them?

image
Also if the load more button is at the bottom of the list, could the "contacts" title in the above image be left aligned so as to be more visible?

**Comment by squigglybob** [comment]
Also for consistency with the UI would it make sense to have a coloured bar next to the left hand side of the records in the dropdown so that it matches the other occasions where colour is used to show the status here, rather than colourising the status text?

image

**Comment by squigglybob** [comment]
Also, and this is partly due to several of the default status colours being very similar, would it be worth having a tooltip when you hover over an item in the typeahead that includes the status name?

**Comment by kodinkat** [comment]
> Firstly, just to say, that I'm loving having the colours of the status's in so many places, it gives some really valuable information to the viewer without using up too much space, and makes it look prettier too :)
> 
> My first thought is that with the colours in all of these typeaheads and dropdowns and such, the contact list is looking a little drab. Knowing that this PR is adding status colours in different places, my first thought was to look at the contact list, but there was no colour 😞 .
> 
> Are there any thoughts to add status colours to the contacts list as well? I know we already have a column which shows by default for the status names for each record, but if it could also be coloured in some way that would also give a nice visual cue on the list. Just a thought.

Hey @squigglybob thanks for the feedback and that's actually a good shout! The corresponding status colour could be added to the left-hand side of the list row.

@corsacca any mileage in working this in, across the various post type lists?

**Comment by kodinkat** [comment]
> Secondly, I have a question about the "load more" of the advanced search. When clicking around to see what did what, I noticed a load more button on the record title section.
> 
> Clicking the button added the next set of contacts to the list but at the top... image
> 
> where Alsha is the first contact that was on the list before I pressed the button, and the next set of contacts have been prepended to the list in reverse alphabetical order.
> 
> Would it make more sense for the load more to be at the bottom of the e.g. contacts record section? And then when you click it, it will be adding the new records to the bottom of the list as well where you would be expecting them?
> 
> image Also if the load more button is at the bottom of the list, could the "contacts" title in the above image be left aligned so as to be more visible?

Hmmm, I think this may come down to a usability consensus, as I've grown accustom to having the `Load More` button at the start of the section, with the results pre-pended.

I guess there is nothing stopping us from having the button at the start and end of a section? However, with regards to appending the results, viewers would need to scroll up to the start of the loaded batch.

**Comment by kodinkat** [comment]
> Also for consistency with the UI would it make sense to have a coloured bar next to the left hand side of the records in the dropdown so that it matches the other occasions where colour is used to show the status here, rather than colourising the status text?
> 
> image
> 
> I wondered about this after I saw that the advanced search has the status text next to the record name, but the status text is not coloured in that instance.

I guess from a uniformity standpoint, that makes senses - Should just be a case of playing around with the relevant typeahead results list template.

**Comment by kodinkat** [comment]
> Also, and this is partly due to several of the default status colours being very similar, would it be worth having a tooltip when you hover over an item in the typeahead that includes the status name?

Funny, as tooltips were mentioned during the initial brief; but seemed to have fallen through the cracks... I guess this is doable and just a case of working it in, assuming typeahead plays nicely! :wink:

**Pull request merged**